### ISSUE TITLE
ERB hyphenated tags do not remove whitespace

### ISSUE THREAD

**Issue by fpsvogel** [bridgetownrb/bridgetown#342]
Hyphenated ERB tags such as `<%- if x -%>` do not remove whitespace, and result in the same output as regular, unhyphenated tags (`<% if x %>`).

**Bridgetown Version**: 0.21

**Example Repo**: https://github.com/fpsvogel/erb-hyphenated-tag-test

**To Reproduce**

1. `bridgetown new erb-hyphenated-tags-test`
2. Rename `index.md` to `index.erb`
3. Replace its content with:
```
<span><%= "beep" -%> beep</span>

<%- 2.times do -%>
  <span>
    boop
  </span>
<%- end -%>
```
4. `yarn start`
5. Remove the hyphens and serve the site again. Load the homepage in a new tab to compare.

**Current behavior**
Whether or not hyphens are included in the ERB tags, the resulting homepage contains "beep beep boop boop" (with a space between each word), and the HTML looks like:
```
<span>beep beep</span>

  <span>
    boop
  </span>
  <span>
    boop
  </span>
```

**Expected behavior**
Using hyphenated tags, there should be no space between the words. Or at least there should be some difference in spacing in the resulting HTML. I'm not 100% sure (newbie here, sorry) whether hyphenated tags should remove all spaces and newlines, or just one adjacent space/newline, but this example covers both cases. Also, I tried just one trailing hyphen `<% ... -%>` instead of two, since I have seen that in some places as well, but that results in the same output as above.

**Computing environment (please complete the following information):**
 - OS: Ubuntu 20.04
 - Browser: Firefox Developer Edition
 - Browser Version: 90.0b12
 - Ruby Version: 3.0.1

**Comment by jaredcwhite** [comment]
@fpsvogel I looked into this—the behavior is exactly the same as in a Rails application. I agree with you, it's a little confusing when/how whitespace removal does occur, but this is how the underlying [Erubi implementation](https://github.com/jeremyevans/erubi) works.

**Comment by fpsvogel** [comment]
@jaredcwhite Thank you for looking into that—I have not gotten into Rails yet, so apologies for the non-issue.

It's too bad that the hyphenated tags behave unintuitively. Maybe it will make sense to me later.

For future reference if anyone else stumbles into this issue: for me, some extra whitespace was resulting in a rendered space only between `inline` and `inline-block` elements, because HTML says line breaks between these elements should be appear as a space. My hope here was to eliminate the whitespace, but now I've found that there are other workarounds, helpfully listed in [this CSS-tricks article](https://css-tricks.com/fighting-the-space-between-inline-block-elements/). I chose to adjust the margins of the affected elements to compensate for the space, and then [make the spaces non-selectable](https://stackoverflow.com/a/6900392/4158773). Now I can happily keep the line breaks and indentation in my ERB code.
